filament/backend/include/backend/Platform.h

```diff
@@ -255,6 +255,34 @@ class UTILS_PUBLIC Platform {
         REALTIME,
     };
 
+    /**
+     * Defines how asynchronous operations are handled by the engine.
+     */
+    enum class AsynchronousMode : uint8_t {
+        /**
+         * Asynchronous operations are disabled. This is the default.
+         */
+        NONE,
+
+        /**
+         * Attempts to use a dedicated worker thread for asynchronous tasks. If threading is not
+         * supported by the platform, it automatically falls back to using an amortization strategy.
+         */
+        THREAD_PREFERRED,
```

Comment:
I feel like the client is able to do the "preferred" logic themselves given that you've given them an API to check the mode's availability.